core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -150,54 +120,29 @@ public AiCoreDeployment forDeploymentByModel(@Nonnull final AiModel model)
    *
    * @param scenarioId The scenario id to be used for AI Core service calls.
    * @return A new instance of the AI Core Deployment.
-   * @throws NoSuchElementException if no running deployment is found for the scenario.
+   * @throws DeploymentResolutionException if no running deployment is found for the scenario.
    */
   @Nonnull
-  public AiCoreDeployment forDeploymentByScenario(@Nonnull final String scenarioId)
-      throws NoSuchElementException {
-    return new AiCoreDeployment(
-        this, () -> DEPLOYMENT_CACHE.getDeploymentIdByScenario(this, resourceGroup, scenarioId));
+  public HttpDestination getDestinationForDeploymentByScenario(
+      @Nonnull final String resourceGroup, @Nonnull final String scenarioId)
+      throws DeploymentResolutionException {
+    val deploymentId = deploymentResolver.getDeploymentIdByScenario(resourceGroup, scenarioId);
+    return toInferenceDestination(resourceGroup, deploymentId);
   }
 
   /**
-   * Get a destination using the default service binding loading logic.
+   * Get an {@link ApiClient} to execute requests based on clients generated from OpenAPI
+   * specifications.
    *
-   * @return The destination.
-   * @throws DestinationAccessException If the destination cannot be accessed.
-   * @throws DestinationNotFoundException If the destination cannot be found.
+   * @return A new client object based on {@link #getBaseDestination()}.
    */
   @Nonnull
-  protected HttpDestination getBaseDestination()
-      throws DestinationAccessException, DestinationNotFoundException {
-    return destinationResolver.getDestination();
-  }
-
-  /**
-   * Get the destination builder with adjustments for AI Core.
-   *
-   * @param destination The destination.
-   * @return The destination builder.
-   */
-  @Nonnull
-  protected DefaultHttpDestination.Builder getDestinationBuilder(
-      @Nonnull final Destination destination) {
-    val builder = DefaultHttpDestination.fromDestination(destination);
-    String uri = destination.get(DestinationProperty.URI).get();
-    if (!uri.endsWith("/")) {
-      uri = uri + "/";
-    }
-    builder.uri(uri + "v2/").property(AI_CLIENT_TYPE_KEY, AI_CLIENT_TYPE_VALUE);
-    return builder;
-  }
+  public ApiClient getApiClient() {
```

Comment:
(Question)

Didn't we find a way to mitigate `ApiClient`?

Reply:
Yes, I think in part that is the reason, but also we want to return something that is already pre-configured with the right Jackson so that it works OOTB

Reply:
Agreed, done ✅